docs/developer/develop-minikube.md

```diff
@@ -0,0 +1,178 @@
+# Develop local with VSCode and Minikube
+
+## Introduction
+This document will describe how to setup a local development environment using VSCode as the IDE,  Minikube as the k8s cluster to run Turbinia and Skaffold to handle the development cycle. 
+
+This setup will provide run time debugging with breakpoints/watches in VSCode as well as hot-reloading of code changes to the live Turbinia deployment without having to rebuild the containers or restart the deployment.
+
+---
+NOTE: This setup has been tested and is in active use by the Turbinia developers in the following configurations.
+* MacAir M3 8GB with Docker Desktop
+* GCE (e2-standard-4) Debian 12/bookworm 16GB with Docker Engine (Running on GCP using the [VSCode Remote SSH extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh))
+   * Setting up the VSCode Remote-SSH extension is outside of scope for this documentation but more information can be found [here](https://code.visualstudio.com/docs/remote/ssh).
+
+## Components
+### Minikube
+As the main supported way to run Turbinia is on k8s it makes sense to also develop as close to a production setup as possible. [Minikube](https://minikube.sigs.k8s.io/docs/) is local Kubernetes cluster so you can run the complete Turbinia stack from your laptop.
+
+### Skaffold
+[Skaffold](https://skaffold.dev/) is a development workflow manager that takes care of building and deploying and monitoring for code changes so you can focus on writing code instead of manually building or restarting containers after every code change.
+
+### VSCode + Google Cloud Code
+VSCode is an IDE that combined with the Google Cloud Code extension will manage the dependencies needed for the development workflow and provide an interface to debug through breakpoints and watches.
+
+---
+## Setup
+### Requirements
+Please install the following requirements on your system.
+* [Docker](https://docs.docker.com/engine/install/), you can install Docker Desktop or only the engine.
```

Comment:
I might be misunderstanding how the GCP parts of this work, but if these are not required in that case, can we add a note for which requirements are needed whether you are doing this locally vs. remote?

Reply:
They are all required. This document describes a non-GKE setup, so a local development setup. Local can be seen here as your own laptop or a GCE machine (which is considered a local machine in the cloud...lol). 
The remote k8s development option running on GKE will be added in follow-up PRs.

Reply:
I see, makes sense.  Could we maybe add a note about this too to make it clear how we're using the GCE instance with minikube there locally?

Reply:
Added note to the introduction section where the GCE machine is mentioned.